core/posts/views.py

```diff
@@ -1,50 +1,120 @@
-from django.shortcuts import render
-from django.core.paginator import Paginator
+import logging
 
+from django.core.paginator import Paginator
 # Create your views here.
-from rest_framework import viewsets
+from rest_framework import viewsets, status
 from rest_framework.decorators import action
 from rest_framework.response import Response
 
+from core.authors.models import Follow
+from core.authors.util import get_author_id
+from core.posts.constants import DEFAULT_POST_PAGE_SIZE
+from core.posts.create_posts_view import handle_posts
 from core.posts.models import Posts, Comments
 from core.posts.serializers import PostsSerializer, CommentsSerializer
-from core.posts.create_posts_view import handle_posts
 from core.posts.util import can_user_view, add_page_details_to_response
-from core.posts.constants import DEFAULT_POST_PAGE_SIZE
-from core.authors.models import Follow
-from core.authors.util import get_author_id
-
-import logging
 
 logger = logging.getLogger(__name__)
 
+COMMENT_NOT_ALLOWED = 'Comment not allowed'
+COMMENT_ADDED = 'Comment Added'
+
+
+def create_comment(request, pk=None):
+    post = Posts.objects.get(pk=pk)
+    if not can_user_view(request.user, post):
+        return Response(status=status.HTTP_403_FORBIDDEN)
+    if post:
+        data = request.data
+        print(data)
+        comment = data.get('comment', None)
+        author = comment.get('author', None)
+        author_id = author['id'].rstrip('/').split('/')[-1]
+        comment['author'] = author_id
+        serializer = CommentsSerializer(data=comment)
+        if serializer.is_valid():
+            post.comments.create(**serializer.validated_data)
+            return Response({
+                "query": "addComment",
+                "success": True,
+                "message": COMMENT_ADDED
+            }, status=status.HTTP_200_OK)
+        else:
+            return Response({
+                "query": "addComment",
+                "success": False,
+                "message": COMMENT_NOT_ALLOWED,
+            }, status=status.HTTP_403_FORBIDDEN)
+    else:
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+def list_comments(request, pk=None):
+    size = int(request.query_params.get("size", 5))
+    queryPage = int(request.query_params.get('page', 0))
+    if size < 1 or queryPage < 0 or size > 100:
+        return Response({
+            "success": False,
+            "message": "The query parameters were invalid",
+            "query": "comments"
+        }, status=status.HTTP_400_BAD_REQUEST)
+
+    post = Posts.objects.get(pk=pk)
```

Comment:
You should handle this post potentially not existing (surround with a try/catch and return status 400 if an error happens)

Reply:
Unless Django already does this by default (idk if it does or not)